File: excelConvert.js
const axios = require('axios');
const fs = require('fs');
const ExcelJS = require('exceljs');

function convertExcelToCSV(url, excelPath = 'afl.xlsx', csvPath = 'afl.csv') {
  return new Promise((resolve, reject) => {
    // Download the Excel file
    axios({
      method: 'get',
      url,
      responseType: 'arraybuffer',
    })
      .then((response) => {
        fs.writeFileSync(excelPath, response.data);

        // Read the Excel file
        const workbook = new ExcelJS.Workbook();
        workbook.xlsx.readFile(excelPath).then(() => {
          const worksheet = workbook.worksheets[0];

          // Convert to CSV
          const csvWriter = workbook.csv.writeBuffer({ sheetId: worksheet.id });
          csvWriter.then((buffer) => {
            const csvContent = buffer.toString();
            const lines = csvContent.split('\n');
            const newCsvContent = lines.slice(1).join('\n'); // Remove the first line

            fs.writeFileSync(csvPath, newCsvContent);
            resolve(`Excel file has been converted to CSV (with the top line removed) and saved as ${csvPath}`);
          });
        });
      })
      .catch((error) => {
        reject(`An error occurred while downloading or converting the file: ${error}`);
      });
  });
}

module.exports = convertExcelToCSV;

File: excel_convert.py
import pandas as pd
import requests

def convert_excel_to_csv(url, excel_path='afl.xlsx', csv_path='afl.csv'):
    response = requests.get(url)
    if response.status_code == 200:
        with open(excel_path, 'wb') as f:
            f.write(response.content)
    else:
        raise Exception(f"Failed to download file: HTTP {response.status_code}")

    df = pd.read_excel(excel_path)
    df.to_csv(csv_path, index=False)
    print(f"Excel file has been converted to CSV and saved as {csv_path}")